pos_fixed_discount/static/src/js/FixedDiscountButton.js

```diff
@@ -17,8 +17,7 @@ odoo.define("pos_fixed_discount.FixedDiscountButton", function (require) {
                 startingValue: 0,
             });
             if (confirmed) {
-                var val = Math.round(parseFloat(payload));
-                this.apply_discount(val);
+                this.apply_discount(payload.replace(",", "."));
```

Comment:
**comment**: Tested this, couldn't find a way for the `replace`, even if I type `,` in the discount, I still get a `.` in the payload. Is it due to language settings in Odoo or the browser?  Either way, this seems fine to me.

I'd perhaps keep the call to `parseFloat` just to be safe with Javascript?

Really what was breaking the breaking the flow was only the call to `Math.round`